rest_sensors.cpp

```diff
@@ -2111,6 +2111,89 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
+                    {
+                        if (sensor->modelId() == QLatin1String("ZHEMI101"))
+                        {
+                            quint16 pulseConfiguration = map[pi.key()].toUInt(&ok);
+
+                            if (ok && addTaskSimpleMeteringReadWriteAttribute(task, deCONZ::ZclWriteAttributesId, 0x0300, deCONZ::Zcl16BitUint, pulseConfiguration, VENDOR_DEVELCO))
+                            {
+                                updated = true;
+                            }
+                            else
+                            {
+                                rsp.list.append(errorToMap(ERR_ACTION_ERROR, QString("/sensors/%1/config/%2").arg(id).arg(pi.key()).toHtmlEscaped(),
+                                                           QString("Could not set attribute")));
+                                rsp.httpStatus = HttpStatusBadRequest;
+                                return REQ_READY_SEND;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        rsp.list.append(errorToMap(ERR_INVALID_VALUE, QString("/sensors/%1/config/%2").arg(id).arg(pi.key()).toHtmlEscaped(),
+                                                   QString("invalid value, %1, for parameter %2").arg(map[pi.key()].toString()).arg(pi.key()).toHtmlEscaped()));
+                        rsp.httpStatus = HttpStatusBadRequest;
+                        return REQ_READY_SEND;
+                    }
+                }
+                else if (rid.suffix == RConfigInterfaceMode)
+                {
+                    if (map[pi.key()].type() == QVariant::Double)
+                    {
+                        if (sensor->modelId() == QLatin1String("ZHEMI101"))
+                        {
+                            quint8 mode = map[pi.key()].toUInt(&ok);
+                            quint16 interfaceMode = 0;
+                            
+                            if (ok && mode == 1) { interfaceMode = 0x0000; }          // Pulse Counting on an Electricity Meter – Unit KWh
+                            else if (ok && mode == 2) { interfaceMode = 0x0001; }     // Pulse Counting on a Gas Meter – Unit m3
+                            else if (ok && mode == 3) { interfaceMode = 0x0002; }     // Pulse Counting on a Water Meter – Unit m3
+                            else if (ok && mode == 4) { interfaceMode = 0x0100; }     // Kamstrup KMP Protocol
+                            else if (ok && mode == 5) { interfaceMode = 0x0101; }     // Not Supported - Linky Protocol
+                            else if (ok && mode == 6) { interfaceMode = 0x0102; }     // DLMS-COSEM - IEC62056-21 mod A
+                            else if (ok && mode == 7) { interfaceMode = 0x0103; }     // P1 Dutch Standard – DSMR 2.3 Version
+                            else if (ok && mode == 8) { interfaceMode = 0x0104; }     // P1 Dutch Standard – DSMR 4.0 Version
```

Comment:
For the `mode` and `interfaceMode` values I'd suggest defines instead hard coded values, since they are used later in this file again and this might be extended in future?

Reply:
There's indeed a 2nd device to follow. I'd probably even go a step further and later see if a device specific map would make sense, as it is used every now and then for other devices/attributes. That could aid in shrinking the code and make it less error prone, I suppose